webui/react/src/components/Badge.tsx

```diff
@@ -33,6 +37,11 @@ const Badge: React.FC<BadgeProps> = ({
     style.backgroundColor = getStateColorCssVar(state);
   } else if (type === BadgeType.Id) {
     classes.push(css.id);
+  } else if (type === BadgeType.Header) {
+    classes.push(css.header);
+  } else if (type === BadgeType.Custom) {
+    style.color = props.fgColor;
+    style.backgroundColor = props.bgColor;
```

Comment:
**block:** this forces colors to migrate into JS (specifically outside of Badge). It would be better to define all the different types and ways of using it but map the colors to themes or css modules.

Reply:
:+1:  I added SlotState to be able to remove the Custome type